src/measure_theory/measure/measure_space.lean

```diff
@@ -2020,6 +2020,19 @@ lemma self_mem_ae_restrict {s} (hs : measurable_set s) : s ∈ (μ.restrict s).a
 by simp only [ae_restrict_eq hs, exists_prop, mem_principal, mem_inf_iff];
   exact ⟨_, univ_mem, s, subset.rfl, (univ_inter s).symm⟩
 
+/-- If two measurable sets are ae_eq then any proposition that is almost everywhere true on one
+is almost everywhere true on the other -/
+lemma restrict_eq_of_ae_eq_of_restrict_eq {s t} (hst : s =ᵐ[μ] t) {p : α → Prop} :
+  (∀ᵐ (x : α) ∂μ.restrict s, p x) → (∀ᵐ (x : α) ∂μ.restrict t, p x) :=
+by simp [measure.restrict_congr_set hst]
+
+/-- If two measurable sets are ae_eq then any proposition that is almost everywhere true on one
+is almost everywhere true on the other -/
+lemma restrict_eq_iff_of_ae_eq {s t} (hst : s =ᵐ[μ] t) {p : α → Prop} :
```

Comment:
About the name: I would choose `ae_restrict_congr_set`, since it is a congr lemma about the set, which gives you an equivalence of expressions of the form usually called `ae_restrict`. And the name would be similar to existing names like `measure.restrict_congr_set`.

Reply:
As mentioned above, I could not find this because there's a bit of a conflict in the library where I was continually looking for `ae_congr` or `ae_eq` and not finding it. It turned out I needed to look for `congr`. So keeping some reference to `ae_eq` would have helped me.

Now that's just me, and I could obviously be convinced otherwise, but just a suggestion. :-)

Reply:
I agree with Remy here: the name `restrict_eq_of_xxx` is definitely misleading since the conclusion is not a `restrict_eq` statement. I second the proposal of `ae_restrict_congr_set` as a better name.

Reply:
Sounds good to me! Will change.